src/CorrelationBasedFeatureSelector.jl

```diff
@@ -0,0 +1,93 @@
+"""
+`CorrelationBasedFeatureSelector` has the following fields:
+
+* `k::Int64` - Select top `k` features with the highest correlation to target
+  variable. You could ignore this by specifying k <= 0. This defaults to 0.
+* `threshold::Float64` - Select features with correlation more than or equal to
+  threshold. To ignore, simply set threshold to 0 (default behavior).
+"""
+mutable struct CorrelationBasedFeatureSelector
+    k::Int64
+    threshold::Float64
+end
+
+function CorrelationBasedFeatureSelector(; k::Int64 = 0, threshold::Float64 = 0.0)
+    CorrelationBasedFeatureSelector(k, threshold)
+end
+
+"""
+    function select_features(selector,
+                             X::DataFrame,
+                             y::Vector;
+                             verbose::Bool=false,
+                             return_val::Bool=false)
+
+Select features based on the importance, which is defined by `selector` to
+target `y`. if `verbose` is true, logs will be printed - this defaults to
+false. If `return_val` is true, this function will return only the feature
+feature names, otherwise, tuple of selected feature names and the
+correlation value are returned.
```

Comment:
I can see the convenience of `return_val` but I think it's less composable than might be ideal. Maybe consider having functions to compute the "quality" of each feature (for each of correllation, chisq, ftest). Then you can supply those as a function rather than symbol in `method::Symbol` field below (if `UnivariateFeatureSelector` idea below makes sense).